cmd/kpod/info.go

```diff
@@ -0,0 +1,202 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"runtime"
+
+	"github.com/docker/docker/pkg/system"
+	"github.com/ghodss/yaml"
+	"github.com/urfave/cli"
+)
+
+var (
+	infoDescription = "display system information"
+	infoCommand     = cli.Command{
+		Name:        "info",
+		Usage:       infoDescription,
+		Description: infoDescription,
+		Flags:       infoFlags,
+		Action:      infoCmd,
+		ArgsUsage:   "",
+	}
+	infoFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "debug, D",
+			Usage: "display additional debug information",
+		},
+		cli.BoolFlag{
+			Name:  "json",
+			Usage: "output as JSON instead of the default YAML",
+		},
+	}
+)
+
+func infoCmd(c *cli.Context) error {
+	info := map[string]interface{}{}
+
+	infoGivers := []infoGiverFunc{
+		storeInfo,
+		hostInfo,
+	}
+
+	if c.Bool("debug") {
+		infoGivers = append(infoGivers, debugInfo)
+	}
+
+	for _, giver := range infoGivers {
+		thisName, thisInfo, err := giver(c)
+		if err != nil {
+			info[thisName] = infoErr(err)
+			continue
+		}
+		info[thisName] = thisInfo
+	}
+
+	var buf []byte
+	var err error
+	if c.Bool("json") {
+		buf, err = json.MarshalIndent(info, "", "  ")
+	} else {
+		buf, err = yaml.Marshal(info)
+	}
+	if err != nil {
+		return err
+	}
+	fmt.Println(string(buf))
+
+	return nil
+}
+
+func infoErr(err error) map[string]interface{} {
+	return map[string]interface{}{
+		"error": err.Error(),
+	}
+}
+
+type infoGiverFunc func(c *cli.Context) (name string, info map[string]interface{}, err error)
+
+// top-level "debug" info
+func debugInfo(c *cli.Context) (string, map[string]interface{}, error) {
+	// lets say number of goroutines
+	info := map[string]interface{}{}
+	info["goroutines"] = runtime.NumGoroutine()
```

Comment:
nit: This doesn't look super-useful in a serverless design that we have for kpod.